docs: add notes on tile distribution and inline comments (#3297)
ROCm/composable_kernel: tutorial/ck_tile/01_naive_gemm/TILE_DISTRIBUTION.md

# Tile Distribution: Mapping Threads to Data

## Overview

**Tile Distribution** describes how each thread in a thread block maps to elements of a block tile. It defines the hierarchical pattern of data distribution across threads, warps, and thread blocks.

## The Problem

Given a block tile of size `MPerBlock × KPerBlock` (e.g., 256×32), we need to determine:

> - Which thread loads which elements?

Reviewer: ```suggestion
- Which threads load which elements.
```

> - How are threads organized into warps?

Reviewer: ```suggestion
- How the threads are organized into warps.
```

> - How many times does each warp repeat its pattern?

Reviewer: ```suggestion
- The number of times each warp repeats its pattern.
```

> - How many elements can each thread load in a single vector instruction?

Reviewer: ```suggestion
- The number of elements each thread can load in a single vector instruction.
```
